new resource "azurerm_maintenance_assignment" (#6713)
terraform-providers/terraform-provider-azurerm: website/docs/r/maintenance_assignment_dedicated_host.html.markdown

}

resource "azurerm_dedicated_host_group" "example" {
  name                        = "example-host-group"
  resource_group_name         = azurerm_resource_group.example.name
  location                    = azurerm_resource_group.example.location
  platform_fault_domain_count = 2
}

resource "azurerm_dedicated_host" "example" {
  name                    = "example-host"
  location                = azurerm_resource_group.example.location
  dedicated_host_group_id = azurerm_dedicated_host_group.example.id
  sku_name                = "DSv3-Type1"
  platform_fault_domain   = 1
}

resource "azurerm_maintenance_configuration" "example" {
  name                = "example-mc"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  scope               = "All"
}

resource "azurerm_maintenance_assignment_dedicated_host" "example" {
  location                     = azurerm_resource_group.example.location
  maintenance_configuration_id = azurerm_maintenance_configuration.example.id
  dedicated_host_id            = azurerm_dedicated_host.example.id
}
```

## Argument Reference

The following arguments are supported:

> * `location` - (Required) Specified the supported Azure location where the resource exists. Changing this forces a new resource to be created.

Reviewer: Typo
```suggestion
* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
```

> * `maintenance_configuration_id` - (Required) Specified the id of Maintenance Configuration Resource. Changing this forces a new resource to be created.

Reviewer: typo / grammar
```suggestion
* `maintenance_configuration_id` - (Required) Specifies the ID of the Maintenance Configuration Resource. Changing this forces a new resource to be created.
```

> * `dedicated_host_id` - (Required) Specified the Dedicated Host ID where the Maintenance Configuration will be assigned to. Changing this forces a new resource to be created.

Reviewer: typo / grammar
```suggestion
* `dedicated_host_id` - (Required) Specifies the Dedicated Host ID to which the Maintenance Configuration will be assigned. Changing this forces a new resource to be created.
```
